fastn-p2p/src/cli/daemon/control.rs

```diff
@@ -0,0 +1,216 @@
+//! Control socket server for handling client requests
+//!
+//! This module handles the Unix domain socket that clients connect to.
+//! It parses JSON requests and coordinates with the P2P layer.
+
+use std::path::PathBuf;
+use tokio::sync::broadcast;
+use tokio::net::UnixListener;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use serde::{Deserialize, Serialize};
+
+use super::{DaemonCommand, DaemonResponse};
+
+/// JSON request format from clients
+#[derive(Debug, Deserialize)]
+struct ClientRequest {
+    /// Request ID for tracking responses
+    id: String,
+    /// Request type: "call", "stream", "reload-identities", "set-identity-state", "add-protocol", "remove-protocol"
+    #[serde(rename = "type")]
+    request_type: String,
+    /// Target peer ID52 (for call/stream requests)
+    #[serde(default)]
+    target: Option<String>,
+    /// Protocol name (for call/stream/add-protocol/remove-protocol requests)
+    #[serde(default)]
+    protocol: Option<String>,
+    /// Identity name (for identity management requests)
+    #[serde(default)]
+    identity: Option<String>,
+    /// Bind alias (for add-protocol/remove-protocol requests)
+    #[serde(default)]
+    bind_alias: Option<String>,
+    /// Online state (for set-identity-state requests)
+    #[serde(default)]
+    online: Option<bool>,
+    /// Request data (varies by protocol and request type)
+    #[serde(default)]
+    data: Option<serde_json::Value>,
+}
+
+/// JSON response format to clients
+#[derive(Debug, Serialize)]
+struct ClientResponse {
+    /// Request ID from original request
+    id: String,
```

Comment:
Lets get rid of this id you have invented, forces people to generate and think of one, when its just a function call we want to act like.